src/Components/Web.Extensions/src/InputFile/InputFile.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Versioning;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    /// <summary>
+    /// A component that wraps the HTML file input element and exposes a <see cref="Stream"/> for each file's contents.
+    /// </summary>
+    public class InputFile : ComponentBase, IInputFileJsCallbacks, IDisposable
```

Comment:
We should have a sample of how this works with validation. Specifically validation for the max file size. We want to make sure it is possible to avoid starting the file upload all together if a file is bigger than what we want to allow.